tests/framework/cli/test_catalog.py

```diff
@@ -67,6 +67,23 @@ def fake_catalog_with_overlapping_factories():
     return config
 
 
+@pytest.fixture
+def fake_catalog_config_with_overwrite():
```

Comment:
What do you mean with "config_with_overwrite"?

Reply:
In this context this means a catalog configuration where one of the explicit entries in the catalog also matches a pattern in the catalog, and the test is ensuring the explicit entry is not overwritten. Perhaps `fake_catalog_config_with_overwritable_dataset()` could be more intuitive? Or `fake_catalog_config_with_resolvable_dataset()`?

Reply:
I think `fake_catalog_config_with_resolvable_dataset` makes more sense to me.